### PR TITLE
Add development guide

### LINKED ISSUE #145: add contribution doc
**Describe the solution you'd like**
dev guide that details make commands:

`make generate`
`make manifests`
`make deploy`
`make test`
...

variables like:
`ERASER_IMG`
`MANAGER_IMG`

- how to deploy to kind cluster

example:
https://azure.github.io/azure-workload-identity/docs/development.html

### PR DESCRIPTION
**Pull request by LeonardHd**
**What this PR does / why we need it**:

Improves the docs to setup a development environment locally and how to use the development tooling.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #145 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by LeonardHd** [comment]
@sozercan I am still unsure how much documentation on the different make targets would be required to get this useful and merged to fix above mentioned issue.

The issue mentions additions to environmental vars (e.g., `ERASER_IMG`) and actual make targets (e.g., `make test`). Any thoughts on this?

@cmaclaughlin any thoughts on this too?

**Review by Joll59 — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -97,12 +97,11 @@ fmt: ## Run go fmt against code.
...
 	go vet ./...
 
-test: manifests generate fmt vet envtest ## Run tests.
+test: manifests generate fmt vet envtest ## Run unit tests.
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test ./... -coverprofile cover.out
 
-# Run e2e tests
 .PHONY: e2e-test
-e2e-test:
+e2e-test: ## Run e2e tests on kind cluster.
```
Comment (sozercan) [inline Makefile]: it doesn't necessarily be on kind, we run kind in our CI but you can run this on any cluster

Comment (LeonardHd) [reply Makefile]: Good point updated this.

**Comment by LeonardHd** [comment]
Have added some improvements:

- Added additional guidance on how to setup a local environment
- Improved the Makefile's docs that can be shown via `make help`

File: docs/DEVELOP.md [hunk 2]
```diff
@@ -1 +1,17 @@
...
+You can either utilize [Codespaces](https://docs.github.com/en/codespaces/overview) or setup a local environment.
+
+## Local Setup
+
+### Prerequisites:
+- [go](https://go.dev/) with version 1.16 or later.
+- [kind](https://kind.sigs.k8s.io/)
+- `make`
+
+### Get thing's running
```
Comment (pmengelbert) [inline docs/DEVELOP.md]: "things", not "thing's"

**Comment by sozercan** [comment]
> The issue mentions additions to environmental vars (e.g., ERASER_IMG) and actual make targets (e.g., make test). Any thoughts on this?

@LeonardHd thanks for the PR! i think there's value in describing the make targets for a new developer to the project.

this is a good example for tests:
https://azure.github.io/azure-workload-identity/docs/development.html

additionally, we can add a detail on `make manifests` for example. example: https://book.kubebuilder.io/reference/generating-crd.html#generating-crds

let me know what you think

**Review by pmengelbert — COMMENTED** [review]
Generally speaking

**Comment by LeonardHd** [comment]
> > The issue mentions additions to environmental vars (e.g., ERASER_IMG) and actual make targets (e.g., make test). Any thoughts on this?
> 
> @LeonardHd thanks for the PR! i think there's value in describing the make targets for a new developer to the project.
> 
> this is a good example for tests: https://azure.github.io/azure-workload-identity/docs/development.html
> 
> additionally, we can add a detail on `make manifests` for example. example: https://book.kubebuilder.io/reference/generating-crd.html#generating-crds
> 
> let me know what you think

I just added a first draft version with commit [7d086ef](https://github.com/Azure/eraser/pull/187/commits/7d086efb1da79fbf20cfc50d088632e8640f25cc) of how I could envision such documentation. Just want to check what you and the team is thinking about such approach. Feedback and thoughts are welcome, and I would continue working on this if my latest changes align with your idea.

**Comment by bdlb77** [comment]
I think it may be good to add the full development workflow of making a change and running it locally.

So Essentially adding a bit more after generating the Manifests:
- showing the steps of building an Image with new changes
- Deploying your local dev image to a running development cluster

**Comment by sozercan** [comment]
@LeonardHd looks like there's a conflict with `Makefile` now

File: docs/DEVELOP.md [hunk 3]
```diff
@@ -1 +1,274 @@
...
+
+- `make build`
+
+    Builds the eraser manager binaries.
+
+- `make run`
+
+    Runs the eraser manager on your local machine.
+
+- `docker-build-manager`
```
Comment (sozercan) [inline docs/DEVELOP.md]: for consistency, can we prefix with `make`? (or not prefix with `make`, either way as long as they are consistent)

Comment (sozercan) [reply docs/DEVELOP.md]: applies to the ones below as well

**Review by sozercan — APPROVED** [review]
quick comment, otherwise LGTM

**Comment by sozercan** [comment]
@LeonardHd thank you!

**Comment by LeonardHd** [comment]
Updated based on the comments. Thanks, was a fun hack week!

**Pull request merged**